pkg/controller/jobframework/setup_test.go

```diff
@@ -99,6 +117,18 @@ func TestSetupControllers(t *testing.T) {
 			},
 			wantEnabledIntegrations: []string{"batch/job"},
 		},
+		"mapper doesn't have ray.io/raycluster when Controllers have been setup, but eventually does": {
+			opts: []Option{
+				WithEnabledFrameworks([]string{"batch/job", "kubeflow.org/mpijob", "ray.io/raycluster"}),
+			},
+			mapperGVKs: []schema.GroupVersionKind{
+				batchv1.SchemeGroupVersion.WithKind("Job"),
+				kubeflow.SchemeGroupVersionKind,
+				// Not including RayCluster
```

Comment:
what adds it?

Reply:
I've updated the test to add the job framework to the mapper directly. 

However, previously, with the `defaultCheckAPIAvailable` function variable we simulate the RayCluster framework becoming available. For context, first we check if the restMapping exists for any given framework, if not, we start the go routine on `waitForAPI()` with the function variable which will return if the restMapping was found. In the test, we overwrite the function variable to eventually return true, saying that the restMapping for the RayCluster framework was found.